drivers/npmx/npmx_shell.c

```diff
@@ -1792,6 +1799,165 @@ static int cmd_adc_meas_take_vbat(const struct shell *shell, size_t argc, char *
+	} else {
+		shell_error(shell, "Error: unable to read timer configuration.");
+	}
+
+	return 0;
+}
+
+static int timer_config_set(const struct shell *shell, size_t argc, char **argv,
+			    npmx_timer_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	char config_name[10];
+	if (config_type == NPMX_TIMER_CONFIG_TYPE_MODE) {
+		strcpy(config_name, "mode");
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_PRESCALER) {
+		strcpy(config_name, "prescaler");
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_COMPARE) {
+		strcpy(config_name, "period");
+	} else {
+		shell_error(shell, "Error: invalid config type value.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing timer %s value.", config_name);
+		return 0;
+	}
+
+	int err = 0;
+	uint32_t input_value = CLAMP(shell_strtoul(argv[1], 0, &err), 0, 0xFFFFFFUL);
+	if (err != 0) {
+		shell_error(shell, "Error: timer %s has to be an integer.", config_name);
+		return 0;
+	}
+
+	npmx_timer_t *timer_instance = npmx_timer_get(npmx_instance, 0);
+
+	npmx_timer_config_t timer_config;
+	npmx_error_t err_code = npmx_timer_config_get(timer_instance, &timer_config);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read timer configuration.");
+		return 0;
+	}
+
+	if (config_type == NPMX_TIMER_CONFIG_TYPE_MODE) {
+		timer_config.mode = timer_mode_int_convert_to_enum(input_value);
+
+		if (timer_config.mode == NPMX_TIMER_MODE_INVALID) {
+			shell_error(shell,
+				    "Error: timer mode can be: 0-boot monitor, 1-watchdog warning, "
+				    "2-watchdog reset, 3-general purpose or 4-wakeup.");
+			return 0;
+		}
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_PRESCALER) {
+		if (input_value < NPMX_TIMER_PRESCALER_COUNT) {
+			timer_config.prescaler = (input_value ? NPMX_TIMER_PRESCALER_FAST :
+								NPMX_TIMER_PRESCALER_SLOW);
+		} else {
+			shell_error(shell, "Error: timer prescaler value can be 0-slow or 1-fast.");
+			return 0;
+		}
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_COMPARE) {
+		timer_config.compare_value = input_value;
+	}
+
+	err_code = npmx_timer_config_set(timer_instance, &timer_config);
+
+	if (check_error_code(shell, err_code)) {
+		if (config_type == NPMX_TIMER_CONFIG_TYPE_MODE) {
+			shell_print(shell, "Success: %d.", timer_config.mode);
+		} else if (config_type == NPMX_TIMER_CONFIG_TYPE_PRESCALER) {
+			shell_print(shell, "Success: %d.", timer_config.prescaler);
+		} else if (config_type == NPMX_TIMER_CONFIG_TYPE_COMPARE) {
+			shell_print(shell, "Success: %d.", timer_config.compare_value);
+		}
+	} else {
+		shell_error(shell, "Error: unable to set timer configuration.");
+	}
+
+	return 0;
+}
+
+static int cmd_timer_config_mode_get(const struct shell *shell, size_t argc, char **argv)
+{
+	return timer_config_get(shell, NPMX_TIMER_CONFIG_TYPE_MODE);
+}
```

Comment:
```suggestion
static int cmd_timer_config_mode_get(const struct shell *shell, size_t argc, char **argv)
{
        ARG_UNUSED(argc);
	ARG_UNUSED(argv);
	return timer_config_get(shell, NPMX_TIMER_CONFIG_TYPE_MODE);
}
```
I would change it just to avoid compile errors/warnings. Also in other functions.